### PR TITLE
bdk-cli basics 00

### PR DESCRIPTION
**Pull request by logosstone**
This is a basics bdk-cli tutorial for quick reference.

### CODE REVIEW

**Comment by rajarshimaitra** [comment]
Hey @waterst0ne is this still WIP?? Sorry took me some time to get back to the docs.. But really excited on this one.. Let me know if this is ready for review and will go through it..

**Comment by logosstone** [comment]
> Hey @waterst0ne is this still WIP?? Sorry took me some time to get back to the docs.. But really excited on this one.. Let me know if this is ready for review and will go through it..

Hi @rajarshimaitra this is ready for review. I have a similar "2-of-3  Multisig" tutorial like this just about finished but I would just have to make some final changes. I'll  upload soon, thank you for reaching out.

File: docs/_blog/bdk_cli_basics_00.md
Comment (rajarshimaitra) [inline docs/_blog/bdk_cli_basics_00.md]: A more verbose title, something like
```suggestion
title: "Command Line introduction to Bitcoin Wallet Development using bdk-cli"
```

File: docs/_blog/bdk_cli_basics_00.md
Comment (rajarshimaitra) [inline docs/_blog/bdk_cli_basics_00.md]: Lets mark the "language bindings" words with hyperlink instead of "BDK"

File: docs/_blog/bdk_cli_basics_00.md
Comment (rajarshimaitra) [inline docs/_blog/bdk_cli_basics_00.md]: How about a hyperlink for bdk-cli pointing to the github repo?

File: docs/_blog/bdk_cli_basics_00.md
Comment (rajarshimaitra) [inline docs/_blog/bdk_cli_basics_00.md]: Same hyperlink love for "Mastering Bitcoin" would be nice too..

File: docs/_blog/bdk_cli_basics_00.md
Comment (rajarshimaitra) [inline docs/_blog/bdk_cli_basics_00.md:11]: This should read something like "Tutorial Goals".. The description for the tutorial is already there in the title.. Also I feel marking it with ## instead # would look more nice in symmetry with other chapter titles.

File: docs/_blog/bdk_cli_basics_00.md
Comment (rajarshimaitra) [inline docs/_blog/bdk_cli_basics_00.md]: Its mostly called as "Xprv"..

File: docs/_blog/bdk_cli_basics_00.md
Comment (rajarshimaitra) [inline docs/_blog/bdk_cli_basics_00.md]: We don't really need `repl` and `compiler` for this tutorial..

File: docs/_blog/bdk_cli_basics_00.md
Comment (rajarshimaitra) [inline docs/_blog/bdk_cli_basics_00.md]: We are currently at 0.6.0..

File: docs/_blog/bdk_cli_basics_00.md
Comment (rajarshimaitra) [inline docs/_blog/bdk_cli_basics_00.md]: A preview of `bdk-cli help` and a bit of intro on each command would be helpful here..

File: docs/_blog/bdk_cli_basics_00.md
Comment (rajarshimaitra) [inline docs/_blog/bdk_cli_basics_00.md]: A note here saying reader need to wait for a 1 block confirmation to see the balance update would be helpful..

File: docs/_blog/bdk_cli_basics_00.md
Comment (rajarshimaitra) [inline docs/_blog/bdk_cli_basics_00.md]: I am confused.. What is this address and why does it look like this??

File: docs/_blog/bdk_cli_basics_00.md
Comment (rajarshimaitra) [inline docs/_blog/bdk_cli_basics_00.md]: Finishing touch, run sync one more time and see your balance decreased..

File: docs/_blog/bdk_cli_basics_00.md
Comment (rajarshimaitra) [inline docs/_blog/bdk_cli_basics_00.md]: One more link for PSBT would be nice..

**Review by rajarshimaitra — CHANGES_REQUESTED** [review]
ACK 35e029fa6105a6a04e27ae5f6187adee594dd2b3

Thanks for making this.. Really looking forward for your multisig tutorial.. If you are interested in writing some other cool ones with more complex scripts ping me in discord.. I have few ideas.. 

This one looks good, but I think it fits better in the tutorial section than in the blog posts.. 

Really like your video gif idea to show the commands in action.. This can be useful for more complex and involved tutorials.. 

For now some following nits..

File: docs/_blog/bdk_cli_basics_00.md
Comment (logosstone) [reply docs/_blog/bdk_cli_basics_00.md]: Good call.

File: docs/_blog/bdk_cli_basics_00.md
Comment (logosstone) [reply docs/_blog/bdk_cli_basics_00.md]: yes.

File: docs/_blog/bdk_cli_basics_00.md
Comment (logosstone) [reply docs/_blog/bdk_cli_basics_00.md]: Yes 👍

File: docs/_blog/bdk_cli_basics_00.md
Comment (logosstone) [inline docs/_blog/bdk_cli_basics_00.md]: Thank you.

File: docs/_blog/bdk_cli_basics_00.md
Comment (logosstone) [reply docs/_blog/bdk_cli_basics_00.md]: That makes sense in case the user wants to tinker . 💯

File: docs/_blog/bdk_cli_basics_00.md
Comment (logosstone) [reply docs/_blog/bdk_cli_basics_00.md]: ACK

File: docs/_blog/bdk_cli_basics_00.md
Comment (logosstone) [reply docs/_blog/bdk_cli_basics_00.md]: I got this address from a test-net faucet and I'm not sure what kind of address is this. It shouldn't interfere with the tutorial in my opinion.

File: docs/_blog/bdk_cli_basics_00.md
Comment (logosstone) [reply docs/_blog/bdk_cli_basics_00.md]: Tip was added!

File: docs/_blog/bdk_cli_basics_00.md
Comment (logosstone) [reply docs/_blog/bdk_cli_basics_00.md]: Added it to the list. 
Than you  for all the recommendations, the tutorial looks much better now.

**Comment by rajarshimaitra** [comment]
Hey @waterst0ne is this ready for a second look? LMK, the new commit seems to be re adding the whole thing, and not a change set.. I tried manually checking few review comments but they don't seem to be update.. Having a change set of updates only would help in review a lot..

**Comment by logosstone** [comment]
> 

Hi Raj, yes this is ready for a second review please. Regarding the commit not showing a change set updates I'll take note, thank you for mentioning it. Feel free to share more suggestions or corrections.

File: docs/tutorials/bdk_cli_basics_00.md
Comment (rajarshimaitra) [inline docs/tutorials/bdk_cli_basics_00.md]: Nit
```suggestion
- The goal for this tutorial is to introduce you to [bdk-cli](https://github.com/bitcoindevkit/bdk-cli), a powerful command-line program used for Bitcoin wallet testing. Made using the more generic [BDK](https://docs.rs/bdk/latest/bdk/) wallet library.  You will be exposed to many of the basic technical skills that go into creating and managing Bitcoin wallets. 
```

File: docs/_blog/bdk_cli_basics_00.md
Comment (rajarshimaitra) [reply docs/_blog/bdk_cli_basics_00.md]: We don't need repl and compiler..

File: docs/_blog/bdk_cli_basics_00.md
Comment (rajarshimaitra) [reply docs/_blog/bdk_cli_basics_00.md]: better to update at v0.6.. Interface changed a bit..

File: docs/tutorials/bdk_cli_basics_00.md
Comment (rajarshimaitra) [inline docs/tutorials/bdk_cli_basics_00.md]: Now I am thinking if this is correct. We should also be able to see unconfirmed balances right? Maybe we can also show how the balance transformed from confirmed to unconfirmed??

File: docs/tutorials/bdk_cli_basics_00.md
Comment (rajarshimaitra) [inline docs/tutorials/bdk_cli_basics_00.md]: With version 6, this balance structure updates.. And we now have a lot more details..

File: docs/tutorials/bdk_cli_basics_00.md
Comment (rajarshimaitra) [inline docs/tutorials/bdk_cli_basics_00.md]: Hmm I think its better to work with old formats than to burn test coins.. I am just thinking if this could confuse readers.. But I guess they could learn something interesting too if they notice it looks weird..

**Review by rajarshimaitra — CHANGES_REQUESTED** [review]
ACK a9c0b21c9999cbc940308cd0afe2a124493e222c. Thanks for the updates.. 

You also need to include the new tutorial path here to get it displayed in the website..
https://github.com/bitcoindevkit/bitcoindevkit.org/blob/689236703f372b1ef82795000a7027f35248b883/docs/.vuepress/config.js#L49-L59 

Took me some time to figure, but you forgot to delete the file from `_blog` folder. That's why previous changes aren't outdated yet.. Can fix that on next push.. 

few more nits, and update requests..

**Comment by thunderbiscuit** [comment]
@waterst0ne are you still working on this? I'm looking to clear up the PR backlog for the website this week. Let me know what you want to do with this.

If it's almost there I can read/fix a bit and merge, otherwise if it's too far back in terms of API we'll just close it.

**Comment by logosstone** [comment]
> @waterst0ne are you still working on this? I'm looking to clear up the PR backlog for the website this week. Let me know what you want to do with this.
> 
> If it's almost there I can read/fix a bit and merge, otherwise if it's too far back in terms of API we'll just close it.

I added some minor changes and also added windows powershell commands.
I think it's ready now.

**Comment by thunderbiscuit** [comment]
I cleaned up a lot of the structure without changing the content and squashed the commits.

**Comment by thunderbiscuit** [comment]
@rajarshimaitra do you mind approving this? The PR can't be merged because you had requested changes but I think it's ready to go now.

**Review by rajarshimaitra — APPROVED** [review]
Sorry took me some time.. 

ACK 7677a3b6ad392dda44047d1594395a9d46f5fc23

**Pull request merged**
